### PR TITLE
Split test artifact plugin into base and conventional plugin

### PR DESCRIPTION
**Pull request by breskeby**
This is a follow up on #68766 which allows reusing test artifact setup
for other sourceSet than `test`.

Also fixes deprecated usage of `internalClusterTestRuntime` which will be removed with Gradle 7.0

### CODE REVIEW

File: x-pack/plugin/repository-encrypted/build.gradle [hunk 1]
```diff
@@ -18,9 +18,9 @@ dependencies {
     // required for integ tests of encrypted FS repository
     internalClusterTestImplementation project(":test:framework")
     // required for integ tests of encrypted cloud repositories
-    internalClusterTestImplementation project(path: ':plugins:repository-gcs', configuration: 'internalClusterTestArtifacts')
-    internalClusterTestImplementation project(path: ':plugins:repository-azure', configuration: 'internalClusterTestArtifacts')
-    internalClusterTestImplementation(project(path: ':plugins:repository-s3', configuration: 'internalClusterTestArtifacts')) {
+    internalClusterTestImplementation testArtifact(project(':plugins:repository-gcs'), 'internalClusterTest')
+    internalClusterTestImplementation testArtifact(project(':plugins:repository-azure'), 'internalClusterTest')
+    internalClusterTestImplementation(testArtifact(project(':plugins:repository-s3'), 'internalClusterTest')) {
```
Comment (mark-vieira) [inline x-pack/plugin/repository-encrypted/build.gradle:23]: nit: can we remove the parens here to be consistent. This is making my brain hurt.

File: plugins/repository-azure/build.gradle [hunk 2]
```diff
@@ -369,17 +370,6 @@ tasks.register("azureThirdPartyTest", Test) {
...
-}
-
-def internalClusterTestJar = tasks.register("internalClusterTestJar", Jar) {
-  appendix 'internalClusterTest'
-  from sourceSets.internalClusterTest.output
-}
-
-artifacts {
-  internalClusterTestArtifacts internalClusterTestJar
+testArtifacts {
```
Comment (mark-vieira) [inline plugins/repository-azure/build.gradle:373]: Can we move this config higher up in the build file? Seems better to not have this oddly buried down here.

**Review by mark-vieira — APPROVED** [review]
Sweet, LGTM.

File: x-pack/plugin/repository-encrypted/build.gradle (hunk 1 above)
Comment (breskeby) [reply x-pack/plugin/repository-encrypted/build.gradle:23]: no we can't as there's a configuration block is following. that doesn't work without the parenthesis due to groovy not supporting this syntax

File: plugins/repository-azure/build.gradle (hunk 2 above)
Comment (breskeby) [reply plugins/repository-azure/build.gradle]: Moved those just below the dependency declarations

File: x-pack/plugin/repository-encrypted/build.gradle (hunk 1 above)
Comment (mark-vieira) [reply x-pack/plugin/repository-encrypted/build.gradle:23]: Ah, my bad, I didn't see that. Ok, this is fine with me then.

**Pull request merged**